### ISSUE TITLE
setup_msvc shadows arch parameter

### ISSUE THREAD

**Issue by shoover** [waf-project/waf#1615]
[msvc.py#L94](https://github.com/waf-project/waf/blob/master/waflib/Tools/msvc.py#L94)

The parameter is shadowed inside the for loop. It defaults to True and works by default, but it does result in an error if autodetecting explicitly with False:

```
def configure(conf):
	conf.load('msvc')
	conf.autodetect(False)
```

The error:

```
Traceback (most recent call last):
  File "C:\Users\Shawn\dev\external\waf\waflib\Scripting.py", line 157, in waf_entry_point
    run_commands()
  File "C:\Users\Shawn\dev\external\waf\waflib\Scripting.py", line 259, in run_commands
    ctx = run_command(cmd_name)
  File "C:\Users\Shawn\dev\external\waf\waflib\Scripting.py", line 243, in run_command
    ctx.execute()
  File "C:\Users\Shawn\dev\external\waf\waflib\Configure.py", line 171, in execute
    super(ConfigurationContext, self).execute()
  File "C:\Users\Shawn\dev\external\waf\waflib\Context.py", line 226, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "C:\Users\Shawn\dev\external\waf\waflib\Context.py", line 304, in recurse
    user_function(self)
  File "C:\Users\Shawn\dev\external\waf\demos\c\wscript", line 22, in configure
    conf.autodetect(False, True)
  File "C:\Users\Shawn\dev\external\waf\waflib\Configure.py", line 345, in fun
    return f(*k, **kw)
  File "C:\Users\Shawn\dev\external\waf\waflib\Tools\msvc.py", line 770, in autodetect
    compiler, version, path, includes, libdirs = conf.detect_msvc(False, lazy)
ValueError: too many values to unpack
```

It could be fixed. Or removed. Or left alone.

**Comment by ita1024** [comment]
We cannot remove the parameter before Waf 1.9. Does renaming the parameter inside the loop fix it? Let us do this if it does.

**Comment by shoover** [comment]
Yes. I will fix it in the loop. It may well be a good candidate to remove
in 1.9.

**Comment by ita1024** [comment]
And I guess this can be closed  77a34692149f487641e3dbb2c242fca2c42602dd
